net/Socket.cpp

```diff
@@ -1002,8 +1002,11 @@ void SocketDisposition::execute()
             _socketMove(_socket);
         } else {
             assert (isTransfer());
-            // Ensure the thread is running before adding callback.
-            _toPoll->startThread();
+            if (!_toPoll->isRunOnClientThread())
+            {
+                // Ensure the thread is running before adding callback.
+                _toPoll->startThread();
```

Comment:
Hmm, I guess the other option is that by the time the SocketDisposition gets its poll (_toPoll), the caller is responsible for starting a thread for the SocketPoll before giving the poll to the SocketDisposition.

I guess no harm to do this here, given that SocketPoll::startThread()  has a counter, so the thread is only started once.

Ah, and SocketDisposition::setTransfer()'s documentation seems to indicate this implicit thread starting will happen.